src/plugins/ui_actions/public/triggers/external_action_trigger.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@osd/i18n';
+import { Trigger } from '.';
+
+export const EXTERNAL_ACTION_TRIGGER = 'EXTERNAL_ACTION_TRIGGER';
+export const externalActionTrigger: Trigger<'EXTERNAL_ACTION_TRIGGER'> = {
```

Comment:
nit - no need to change, but I think the naming here is a little overly generic compared to the description. It's also a little weird to me that the title hard codes the interaction - I don't see why an external action (such as opening a flyout requires a single click, as opposed to a double click or some hotkey combo.

Reply:
Apologies, this action was actually added in #3777 , they show as changes here since I hadn't rebased. You may ignore that part, but if you have more concerns for that, I can try to address that after rebasing.